R/parse.R

```diff
@@ -205,15 +207,23 @@ ensure_valid_pd <- function(pd) {
 #' @param pd A parse table.
 #' @param text The initial code to style.
 #' @keywords internal
-identify_insufficiently_parsed_strings <- function(pd, text) {
+is_insufficiently_parsed_strings <- function(pd) {
   is_problematic_string <- pd$token == "STR_CONST"
   candidate_substring <- substr(
-    pd$text[is_problematic_string], 1, 1
+    pd$text[is_problematic_string], 1L, 1L
   )
   is_problematic_string[is_problematic_string] <- candidate_substring == "["
   is_problematic_string
 }
 
+is_insufficiently_parsed_num_const <- function(pd) {
+  is_problematic_num_const <- pd$token == "NUM_CONST"
+  candidate_substring <- substr(
+    pd$text[is_problematic_num_const], 1L, 2L
+  )
+  is_problematic_num_const[is_problematic_num_const] <- candidate_substring == "0x"
```

Comment:
A regex is shorter and most likely runs faster than creating and checking substrings:

```suggestion
  is_problematic_num_const[is_problematic_num_const] <- grepl("^0x", pd$text[is_problematic_num_const])
```

Same for `..._strings()` above.

Reply:
Right...